src/array_api_stubs/_draft/array_object.py

```diff
@@ -145,8 +154,9 @@ def __abs__(self: array, /) -> array:
         .. versionchanged:: 2022.12
             Added complex data type support.
         """
+        ...
 
-    def __add__(self: array, other: Union[int, float, array], /) -> array:
+    def __add__(self: Self, other: int | float | Self, /) -> Self:
```

Comment:
What the rationale for the avoidance of `Union` and `Optional` in this PR?

Reply:
Neither are necessary with Python's newer language features. I think they're essentially considered legacy, in these contexts, like `typing.Tuple` is obviated by just `tuple`.

Reply:
Given that some of these features are Python 3.10, I suppose we may need to be mindful of backward compat and specifying the minimum supported Python version.

Reply:
I verified that that's correct - this is Python 3.8 compatible. Resolving.